### ISSUE TITLE
abap2xlsx create worksheet how to set fields in header line without filtersymbol? 

### ISSUE THREAD

**Issue by heinkroft** [abap2xlsx/abap2xlsx#325]
Hello Forum,

I am using a package of abap2xlsx (Excel Export ABAP 2 XLSX)   and some methods from zcl_excel classes to create a worksheet for internal table gt_table and download it.

 Following is some Koding I am using,

When the excel worksheet is downloaded, the header line contains filter symbols in all header fields, when using method bind_table (see A:).

Is there a way to set the header line fields without a filtersymbol?

 if I replace the call on method bind_table (see A:)  by call on method convert (see B:)

then there are no filtesymbols, but in this method I cannot give a fieldcatalog table as exporting parameter.

And I want to change the texts in the header.

 Thanks for any suggestions,

 Regards Henricus Kroft

DATA: lo_excel         TYPE REF TO zcl_excel,     

```
       lo_converter    TYPE REF TO zcl_excel_converter,

       lo_worksheet  TYPE REF TO zcl_excel_worksheet,

       lo_writer          TYPE REF TO zif_excel_writer,
```

  Data:

gt_rawdata      TYPE solix_tab.

ls_table_set-table_style  = zcl_excel_table=>builtinstyle_light1.

```
lo_worksheet = lo_excel->get_active_worksheet( ).
```

   A begin:   lo_worksheet->bind_table( ip_table    = gt_table

```
                      it_field_catalog  = lt_field_catalog

                      is_table_settings = ls_table_set ).
```

  A end

```
B begin:   lo_converter->convert( EXPORTING

                            it_table     = gt_table
                            i_row_int    = 1

                            i_column_int = 1

                            io_worksheet = lo_worksheet

                         CHANGING

                            co_excel     = lo_excel ) .
```

B end
- " create output file in EXCEL format as type RAW
  
  lf_xdata = lo_writer->write_file( lo_excel ).
- After 6.40 via cl_bcs_convert
  
  gt_rawdata = cl_bcs_convert=>xstring_to_solix( iv_xstring  = lf_xdata ).
  
  lf_bytecount = xstrlen( lf_xdata ).
  
  OPEN DATASET gf_datn1_str FOR OUTPUT IN BINARY MODE.
  
  CHECK sy-subrc = 0.
  
  lf_bytes_left = lf_bytecount.
  
  LOOP AT gt_rawdata ASSIGNING <ls_rawdata>.
  
    AT LAST.
  
  ```
  CHECK lf_bytes_left >= 0.
  
  TRANSFER <ls_rawdata> TO gf_datn1_str LENGTH lf_bytes_left.
  
  EXIT.
  ```
  
    ENDAT.
  
    TRANSFER <ls_rawdata> TO gf_datn1_str.
  
    SUBTRACT 255 FROM lf_bytes_left.  " Solix hat LÃ¤nge 255
  
  ENDLOOP.
  
  CLOSE DATASET gf_datn1_str.

**Comment by ivanfemia** [comment]
Feature is missing.

**Comment by ivanfemia** [comment]
Enhancement provided please update

**Comment by heinkroft** [comment]
Hello Ivan,
Thank you for your solution. 
 
At the moment I have some other things to do,
 but as soon as I get the time I will check it and let you know.
Kind regards
Hein Kroft  

On Wednesday, August 27, 2014 9:26 PM, Ivan notifications@github.com wrote:

Closed #325.
